packages/@webex/plugin-meetings/src/multistream/mediaRequestManager.ts

```diff
@@ -91,15 +103,64 @@ export class MediaRequestManager {
     this.slotsActiveInLastMediaRequest = activeSlots;
   }
 
+  public setDegradationPreferences(degradationPreferences: DegradationPreferences) {
+    this.degradationPreferences = degradationPreferences;
+    this.sendRequests(); // re-send requests after preferences are set
+  }
+
+  private getDegradedClientRequests() {
+    const clientRequests = cloneDeep(this.clientRequests);
+    const maxFsLimits = [
+      getMaxFs('best'),
+      getMaxFs('large'),
+      getMaxFs('medium'),
+      getMaxFs('small'),
+      getMaxFs('very small'),
+      getMaxFs('thumbnail'),
+    ];
+
+    // reduce max-fs until total macroblocks is below limit
+    for (let i = 0; i < maxFsLimits.length; i += 1) {
+      let totalMacroblocksRequested = 0;
+      Object.values(clientRequests).forEach((mr) => {
+        if (mr.codecInfo) {
+          mr.codecInfo.maxFs = Math.min(
+            mr.codecInfo.maxFs || CODEC_DEFAULTS.h264.maxFs,
+            maxFsLimits[i]
+          );
+          totalMacroblocksRequested += mr.codecInfo.maxFs * mr.receiveSlots.length;
+        }
+      });
+      if (
+        totalMacroblocksRequested <= this.degradationPreferences.maxMacroblocksLimit ||
+        i === maxFsLimits.length - 1
```

Comment:
suggestion: I think you can simplify this a bit and avoid having the extra `|| i === maxFsLimits.length - 1` here if you move the block from lines 143-147 to be in the else clause to this condition from line 134, something like this:
```
  if (totalMacroblocksRequested <= this.degradationPreferences.maxMacroblocksLimit) {
    if (i !== 0) {
      LoggerProxy.logger.warn(
        `multistream:mediaRequestManager --> too many requests with high max-fs, frame size will be limited to ${maxFsLimits[i]}`
      );
    }
    break;
  } else {
    if (i === maxFsLimits.length - 1) {
      LoggerProxy.logger.warn(
        `multistream:mediaRequestManager --> even with frame size limited to ${maxFsLimits[i]} you are still requesting too many streams, consider reducing the number of requests`
      );
    }
  }
```
(I've also changed the log message slightly)